src/histbuf.rs

```diff
@@ -108,22 +101,36 @@ where
     }
 }
 
-impl<T, N> HistoryBuffer<T, N>
-where
-    N: ArrayLength<T>,
-{
+impl<T, const N: usize> HistoryBuffer<T, N> {
+    /// Returns the current fill level of the buffer.
+    #[inline]
+    pub fn len(&self) -> usize {
+        if self.filled {
+            N
+        } else {
+            self.write_at
+        }
+    }
+
     /// Returns the capacity of the buffer, which is the length of the
     /// underlying backing array.
-    pub fn len(&self) -> usize {
-        self.data.len()
+    #[inline]
+    pub fn capacity(&self) -> usize {
+        N
     }
 
     /// Writes an element to the buffer, overwriting the oldest value.
     pub fn write(&mut self, t: T) {
-        self.data[self.write_at] = t;
+        if self.filled {
+            // Drop the old before we overwrite it.
+            unsafe { ptr::drop_in_place(self.data[self.write_at].as_mut_ptr()) }
```

Comment:
For all uses of unsafe, what is the heapless policy regarding docs justifying correctness?

Just curious

Reply:
It currently seems like there is none, only functions with unsafe interfaces are currently documented.
Adding a motivation to all accesses would be good though, but I think that will have to be its own PR.